File: 백준/Silver/1343. 폴리오미노/폴리오미노.py
string = input()
result = []

count = 0
length = len(string)

for i in range(length):
  if string[i] != '.':
    count += 1
  
  if string[i] == '.' or i == length - 1:
    if count % 2 != 0:
      count = -1
      break
    else:
      while count > 0:
        if count - 4 >= 0:
          result.append("AAAA")
          count -= 4
        else:
          result.append("BB")
          count -= 2

      if string[i] == '.':
        result.append(".")

if count >= 0:
  for item in result:
    print(item, end = '')

else:
  print(count)

File: 백준/Silver/1343. 폴리오미노/폴리오미노.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   char str[51];
   char result[51] = "";
   scanf("%s", str);

   int count = 0;
   int length = strlen(str);

   for(int i = 0; i < length; i++){
      if(str[i] != '.'){
         count++;
      }
      if(str[i] == '.' || i == length - 1){
         if(count % 2 == 1){
            printf("-1");
            return 0;
         }
         else{
            while(count > 0){
               if(count - 4 >= 0){
                  strcat(result, "AAAA");
                  count -= 4;
               }
               else{
                  strcat(result, "BB");
                  count -= 2;
               }
            }
            if(str[i] == '.'){
               strcat(result, ".");
            }
         } 
      }
   }

   printf("%s", result);
   
   return 0;
}